lib/Index/IndexRecord.cpp

```diff
@@ -406,12 +406,14 @@ static void addModuleDependencies(ArrayRef<ModuleDecl::ImportedModule> imports,
       case FileUnitKind::Builtin:
         break;
       case FileUnitKind::SerializedAST:
+      case FileUnitKind::DWARFModule:
       case FileUnitKind::ClangModule: {
         auto *LFU = cast<LoadedFile>(FU);
         if (auto *F = fileMgr.getFile(LFU->getFilename())) {
           std::string moduleName = mod->getNameStr();
           bool withoutUnitName = true;
-          if (FU->getKind() == FileUnitKind::ClangModule) {
+          if (FU->getKind() == FileUnitKind::ClangModule ||
+              FU->getKind() == FileUnitKind::DWARFModule) {
```

Comment:
There's a `cast` below that won't work for a DWARFModule.

Reply:
Good catch!